docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,359 @@
+# Design Proposal - Embedding Ingestion Pipeline And RAG-Based Chat
+**TODOs**: 
+* Vector store authentication options.
+* Document versioning and data update policies.
+* Unify prompt management in InstructLab. See (`chat_template` [configuration][chat_template] and
+`augment_chat_template` [function][augment_chat_template])
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1
+
+## 1. Introduction
+This document proposes enhancements to the `ilab` CLI to support workflows utilizing Retrieval-Augmented Generation
+(RAG) artifacts within `InstructLab`. The proposed changes introduce new commands and options for the embedding ingestion
+and RAG-based chat pipelines:
+* A new `ilab data` sub-command to process customer documentation.
+  * Either from knowledge taxonomy or from actual user documents.
+* A new `ilab data` sub-command to generate and ingest embeddings from pre-processed documents into a configured vector store.
+* An option to enhance the chat pipeline by using the stored embeddings to augment the context of conversations, improving relevance and accuracy.
+
+### 1.1 User Experience Overview
+The commands are tailored to support diverse user experiences, all enabling the use of RAG functionality to enrich chat sessions.
+
+### 1.2 Model Training Path
+This flow is designed for users who aim to train their own models and leverage the source documents that support knowledge submissions to enhance the chat context:
+![model-training](../images/rag-model-training.png)
```

Comment:
I like these images as-is, but they do appear to violate the official dev-docs guidelines on images as [specified here](https://github.com/instructlab/dev-docs/blob/main/README.md#images).   My inclination is to leave them as is since they look good, but if the oversight committee decides to be very strict about this guideline then we might need to redo them.

Reply:
Images were actually generated from Excalidraw, one of the recommended tool, and I also added a link to edit them for maintenance and sharing purposes: it's in another section below this paragraph, I can move it up if needed. 
Could you clarify what is the violation that you see in the guidelines? (I had already read them before, that's why I used this tool, BTW)

Reply:
@jwm4 @anastasds please review the "Options to Rebuild Excalidraw Diagrams:" section at the top of the document.